src/IssueViz.UnitTests/Editor/QuickActions/QuickFixes/QuickFixActionsSourceTests.cs

```diff
@@ -93,14 +94,108 @@ public void OnTagsChanged_DismissLightBulbSession()
+                    {
+                        callbackAction();
+                    }
+                    finally
+                    {
+                        eventHandle.Set(); // signal the test that the action has finished
+                    }
+                });
+
+            var issueLocationsTagAggregator = new Mock<ITagAggregator<IIssueLocationTag>>();
+
+            var lightBulbBroker = new Mock<ILightBulbBroker>();
+            lightBulbBroker
+                .Setup(x => x.DismissSession(textView))
+                .Throws(new NotImplementedException("this is a test"));
+
+            CreateTestSubject(issueLocationsTagAggregator.Object, lightBulbBroker.Object, threadHandling: threadHandling.Object);
+
+            lightBulbBroker.VerifyNoOtherCalls();
+
+            Action act = () =>
+            {
+                try
+                {
+                    issueLocationsTagAggregator.Raise(x => x.TagsChanged += null, new TagsChangedEventArgs(Mock.Of<IMappingSpan>()));
+                }
+                finally
+                {
+                    eventHandle.WaitOne();
+                }
+            };
+            act.Should().NotThrow();
+
+            lightBulbBroker.Verify(x => x.DismissSession(textView), Times.Once);
+        }
+
+        [TestMethod]
+        public void OnTagsChanged_CriticalException_ExceptionIsNotCaught()
+        {
+            var eventHandle = new EventWaitHandle(false, EventResetMode.ManualReset);
+
+            var threadHandling = new Mock<IThreadHandling>();
+            threadHandling
+                .Setup(x => x.RunOnUIThread(It.IsAny<Action>()))
+                .Callback((Action callbackAction) =>
+                {
+                    try
+                    {
+                        callbackAction();
+                    }
+                    finally
+                    {
+                        eventHandle.Set(); // signal the test that the action has finished
+                    }
+                });
+
+            var issueLocationsTagAggregator = new Mock<ITagAggregator<IIssueLocationTag>>();
+
+            var lightBulbBroker = new Mock<ILightBulbBroker>();
+            lightBulbBroker
+                .Setup(x => x.DismissSession(textView))
+                .Throws(new StackOverflowException("this is a test"));
+
+            CreateTestSubject(issueLocationsTagAggregator.Object, lightBulbBroker.Object, threadHandling: threadHandling.Object);
+
+            lightBulbBroker.VerifyNoOtherCalls();
+
+            Action act = () =>
+            {
+                try
+                {
+                    issueLocationsTagAggregator.Raise(x => x.TagsChanged += null, new TagsChangedEventArgs(Mock.Of<IMappingSpan>()));
+                }
+                finally
+                {
+                    eventHandle.WaitOne();
+                }
+            };
+            act.Should().ThrowExactly<StackOverflowException>().And.Message.Should().Be("this is a test");
```

Comment:
This is not being thrown, I suspect because this is async. When I'm debugging I can see the exception, but it's not being propagated here. Not sure what I'm doing wrong.

Reply:
@rita-sonarsource have a look at the test “
EventHandler_CriticalException_IsNotSuppressed” that I added in the
keep-alive code a few days ago. The exception wasn’t being propagated to
the test, so I split most of the code into a separate async method that I
could test directly. It isn’t ideal, but makes it testable.
 Is it a similar problem here?